FACES-3328 PrimeFaces markup remains after SPA navigation in Liferay 7.0+

stiemannkj1 opened the pull request:
@ngriffin7a, merging. Please review when you have time.

stiemannkj1 committed:
FACES-3328 PrimeFaces markup remains after SPA navigation in Liferay 7.0+

ngriffin7a commented on bridge-ext/src/main/java/com/liferay/faces/bridge/ext/config/internal/LiferayPortletConfigParam.java:
```diff
@@ -54,7 +54,23 @@
+
+	// Public Constants
+	public static final String NO_DEFAULT_VALUE =
+		"com.liferay.faces.bridge.ext.primeFacesCSSDefaultDataSennaTrackValue.noDefaultValue";
```
@stiemannkj1 Having an init-param name be prefixed with "com.liferay.faces.xyz" is very ordinary, but not for the value. Can `NO_DEFAULT_VALUE` be something simpler like an empty string? What are the possible values other than `permanent`?

ngriffin7a commented on bridge-ext/src/main/java/com/liferay/faces/bridge/ext/renderkit/html_basic/internal/ResourceRendererLiferayImpl.java:
```diff
@@ -64,15 +70,37 @@ public ResourceRendererLiferayImpl() {
 		this.renderHeadResourceIds = renderHeadResourceIds;
 	}
 
+	/* package-private */ static NameValuePair<String, Object> getDataSennaTrack(UIComponent componentResource, boolean isCSS,
```
@stiemannkj1 The `is` prefix is typically limited to boolean getter methods, not parameter names. Should be `css` or `styleSheet` instead of `isCss`

ngriffin7a commented on bridge-ext/src/test/java/com/liferay/faces/bridge/ext/renderkit/html_basic/internal/UICommandMockImpl.java:
```diff
@@ -25,21 +25,31 @@
-	private Map<String, Object> passThroughAttrs;
-	private String rendererType;
+	private final String componentFamily;
+	private final Map<String, Object> passThroughAttrs;
```
@stiemannkj1 Might be a little silly for a mock, but I think we decided that `final Map` would always be unmodifiable.

ngriffin7a commented:
@stiemannkj1 Finished code review. See the 3 minor comments.

stiemannkj1 commented:
@ngriffin7a, I believe I've addressed your concerns in this code review commit: https://github.com/liferay/liferay-faces-bridge-ext/commit/74383cdfc320e9a8ea7a46d65f92d77e2ffd5547. Please check it out and let me know.

ngriffin7a commented:
@stiemannkj1 Looks good. The only change I might suggest is `primeFacesCSSDataSennaTrackDefaultValue` instead of `primeFacesCSSDefaultDataSennaTrackValue` but I'll let you make that call.
